### ISSUE TITLE
Show WCIF Groups on a competition tab

### ISSUE THREAD

**Issue by saranshgrover** [thewca/worldcubeassociation.org#6758]
Most competitions use Groupifier, which updates groups for all events on WCIF. Therefore we have all the data needed to show groups when this is the case, and we don't have to navigate competitors to random PDFs or external websites.

Proposed Solution: When WCIF has groups, add a "Groups" tab on the competition site with either a. individualized group information for the competitor, and optionally b. A way to look at all other groups.

P.S - if there is room for this to be implemented in React - I have most of the code already available [here](https://github.com/saranshgrover/nodus/blob/master/client/src/components/Groups/Groups.tsx)

**Comment by gregorbg** [comment]
I understand that this is an appealing feature for Organisers. But it monopolises Groupifier, which goes against the core idea of WCIF. In Europe (at least before the pandemic hit) a significant portion of competitions used other third-party group assignment tools, and those would be left out if we opt for Groupifier's specific format.

We could of course strive to support as many different grouping tools/formats as possible, but this suddenly creates a lot more work. I'd rather encourage organisers to rely on Groupifier itself more.

**Comment by jonatanklosko** [comment]
@gregorbg Groupifier adds some custom data to WCIF to persist user configuration, but the groups format only uses the WCIF standard. Generally the schedule has Rooms with event Activities like `333-r1`, and then groups are represented by `childActivities` with like `333-r1-g1`. Then people have "competitor" Assignments that point to a specific group activity.

**Comment by gregorbg** [comment]
Huh, never knew. Thanks Jonatan, that might deal with the "controversial" label very quickly.
How are these "competitor" assignments being generated? Is it the third-party tools like Groupifier that patch them into the WCIF?

**Comment by jonatanklosko** [comment]
Yeah, a third-party tool like that should create the group activities inside schedule and then add competitors assignments (which is essentially the grouping process). Then it would make a patch request to save the updated WCIF back to the WCA website.

**Comment by gregorbg** [comment]
Cool, makes sense, thanks!

**Comment by coder13** [comment]
This issue opens discussion how what the best way to view groups is.

I've been plagued by how to best show schedules of rounds of groups for different venues / rooms combinations. DO you first have the user select the venue, and then the room? Do you have them view their groups by round? How do you distinguish between 3x3 round 1 on the "red stage" and "blue stage".
I've created a website (https://competition-groups.surge.sh/) to try doing this. Right now the only page I'm happy with is the page to view an individuals group assignments. The "schedule" page I should prompt the user to at least filter the groups by round instead of showing a long list of all the groups.

While there is the classic understanding of group assignments that we often see on nametags with the group assignment for each first round event, we'd need to think long term about what it would like to also communicate round 2 and round 3 group assignments.
